From d6423417778d049b07a3091c3972160be99c9173 Mon Sep 17 00:00:00 2001
From: Lucas De Marchi <lucas.demarchi@profusion.mobi>
Date: Fri, 7 Sep 2012 14:59:35 -0300
Subject: [PATCH] testsuite: Fix double definition of 64-bits variant

If _FILE_OFFSET_BITS is defined we should not be wrapping these 64
variants, since they are macros in libc.
---
 testsuite/path.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/testsuite/path.c b/testsuite/path.c
index 5cb17bb..c13ff4b 100644
--- a/testsuite/path.c
+++ b/testsuite/path.c
@@ -184,16 +184,20 @@ WRAP_2ARGS(FILE*, NULL, fopen, const char*);
 WRAP_2ARGS(int, -1, mkdir, mode_t);
 WRAP_2ARGS(int, -1, access, int);
 WRAP_2ARGS(int, -1, stat, struct stat*);
-WRAP_2ARGS(int, -1, stat64, struct stat64*);
 WRAP_2ARGS(int, -1, lstat, struct stat*);
+#ifndef _FILE_OFFSET_BITS
+WRAP_2ARGS(int, -1, stat64, struct stat64*);
 WRAP_2ARGS(int, -1, lstat64, struct stat64*);
+WRAP_OPEN(64);
+#endif
 
 WRAP_OPEN();
-WRAP_OPEN(64);
 
 #ifdef HAVE___XSTAT
 WRAP_VERSTAT(__x,);
-WRAP_VERSTAT(__x,64);
 WRAP_VERSTAT(__lx,);
+#ifndef _FILE_OFFSET_BITS
+WRAP_VERSTAT(__x,64);
 WRAP_VERSTAT(__lx,64);
 #endif
+#endif
-- 
1.7.12

